### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by seanyasno**
Finished part A

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import {TodoDatabase, updateTodosView, addTodo, initTodoView} from "./models/index.js";
+
+let todoDatabase = new TodoDatabase(updateTodosView);
```
Comment (ETHANTALJAFFE) [inline src/index.js:3]: use `const` instead of `let` to indicate that the variable does not change

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+window.onload = () => {
+  document.getElementById('add-todo-button').addEventListener('click', () => addTodo(todoDatabase, initTodoView));
+  document.getElementById('todo-input').addEventListener('keyup', (event) => {
+    if (event.keyCode === 13) {
```
Comment (jafold) [inline src/index.js]: create `CONST ENTER_KEY_CODE = 13`

File: src/index.js (hunk 1 above)
Comment (jafold) [inline src/index.js]: use `const` instead of `let` to indicate that the variable does not change

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import {TodoDatabase, updateTodosView, addTodo, initTodoView} from "./models/index.js";
```
Comment (jafold) [inline src/index.js:1]: use single quotes for imports

File: src/models/addTodo.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {TodoItem} from "./index.js";
+import {todoItemView} from "../views/index.js";
+
+export default function addTodo(todoDatabase, initTodoView) {
+  let description = document.getElementById('todo-input').value;
```
Comment (jafold) [inline src/models/addTodo.js]: `const`

File: src/models/addTodo.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  if (description) {
+    let todoItem = new TodoItem(description);
```
Comment (jafold) [inline src/models/addTodo.js]: `const`

File: src/models/TodoItem.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export default class TodoItem {
+  constructor(description='', done=false) {
```
Comment (jafold) [inline src/models/TodoItem.js]: spacing after `=`

File: src/models/TodoDatabase.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    this.todos = this.todos.filter(todo => todo.id !== todoId);
+    if (this.removeTodoEvent)
+      this.removeTodoEvent(...args);
+  }
+
+  checkTodo(todoIndex, ...args) {
+    this.todos = this.todos.map((todo, index) => index === todoIndex ? todo.done = !todo.done : '');
+    if (this.checkTodoEvent)
+      this.checkTodoEvent(...args);
+  }
```
Comment (jafold) [inline src/models/TodoDatabase.js]: why do you use args? You're always passing the `TodoItemView` as 2nd parameter. This ambiguity does not seem necessary and makes it tougher to understand your code flow

File: src/models/todo-actions/index.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import removeTodo from "./removeTodo.js";
+import checkTodo from "./checkTodo.js";
+
+export {
+  removeTodo,
+  checkTodo
+}
```
Comment (jafold) [inline src/models/todo-actions/index.js]: Move module out of the /models directory

File: src/models/todo-actions/todoAction.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+export default function todoAction(effect, elementId) {
+  let todosElements = document.getElementById('todos').children;
+  for (let i = 0; i < todosElements.length; i++) {
+    let todoId = todosElements[i].id.split('-')[2];
```
Comment (jafold) [inline src/models/todo-actions/todoAction.js]: `const`

File: src/models/todo-actions/todoAction.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export default function todoAction(effect, elementId) {
```
Comment (jafold) [inline src/models/todo-actions/todoAction.js]: I don't know if `effect` is the best name. Maybe `callback`?

**Review by jafold — CHANGES_REQUESTED** [review]
Make sure to fix issues throughout the PR. I might have not mentioned the mistake every time it happened. Once should be enough.

Overall pretty cool, good work :)

**Review by jafold — CHANGES_REQUESTED** [review]
One thing: project structure is still not good enough. Modules doesn't make sense. You're updating views and changing states. What does that have to do with `modules`?

File: src/modules/todo-actions/todoAction.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+export default function todoAction(effect, elementId) {
```
Comment (ETHANTALJAFFE) [inline src/modules/todo-actions/todoAction.js:1]: Change to `addEventListenerToItem`

File: src/modules/todo-actions/todoAction.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+  const todosElements = document.getElementById('todos').children;
+  for (let i = 0; i < todosElements.length; i++) {
+    const todoId = todosElements[i].id.split('-')[2];
+    document.getElementById(`${elementId}${todoId}`).addEventListener('click', () => {
+      effect(todoId);
+    });
+  }
```
Comment (jafold) [inline src/modules/todo-actions/todoAction.js:8]: Change to `addEventListenerToItem`

File: src/modules/todo-actions/todoAction.js (hunk 11 above)
Comment (jafold) [inline src/modules/todo-actions/todoAction.js:1]: Change `effect` to `callback`

File: src/modules/todo-actions/checkTodo.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+import todoAction from "./todoAction.js";
+
+export default function checkTodo(todoDatabase) {
+  todoAction((todoId) => {
+    todoDatabase.todos.map(todo => todo.id === Number(todoId) ? todo.done = !todo.done : '');
+  }, 'todo-item-check-');
+}
```
Comment (jafold) [inline src/modules/todo-actions/checkTodo.js:7]: rename it `addCheckEventToItem`

File: src/modules/todo-actions/removeTodo.js [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import todoAction from "./todoAction.js";
+
+export default function removeTodo(todoDatabase) {
```
Comment (jafold) [inline src/modules/todo-actions/removeTodo.js:3]: rename it `addRemoveEventToItem`

**Review by jafold — APPROVED** [review]
Fix issues in next PR

**Pull request merged**